spec/spec.tex

```diff
@@ -1024,6 +1024,45 @@ \subsubsection{Format Strings}
 \item \verb|\'| : Single quote
 \end{itemize}
 
+\subsection{Verification}
+
+To facilitate simulation, model checking and formal methods, there are three non-synthesizable verification statements available: assert, assume and cover. Backends are free to generate the corresponding model checking constructs in the target language, but this is not required by the FIRRTL specification. For example, the System Verilog emitter produces System Verilog assert, assume and cover statements. However, backends should not generate any new synthesizable hardware when handling to verification statements. Backends that do not obey this rule should issue a warning.
```

Comment:
> However, backends should not generate any new synthesizable hardware when handling to verification statements. Backends that do not obey this rule should issue a warning.

I am not sure I understand this line. Do we need this?
Something like @albert-magyar 's GoldenGate compiler might actually generate HW in order to check assertions on the FPGA. I don't think that he needs to emit a warning for that.

On the other hand, we might want to require that backends that do not support these statements should issue a warning and then ignore them.

Reply:
I think that we should establish a contract that formal statements should have no effect on synthesis, in order to avoid [Heisenbugs](https://en.wikipedia.org/wiki/Heisenbug). Any backend that needs/wants to violate that contract should be loud and explicit about it to prevent surprises.

The idea to have backends that don't support them issue a warning is good: I've now added that for the Verilog emitter, I'll add it to the spec too.

Reply:
When would a system ever add undesired synthesizable hardware?
How would that possibly cause a heisenbug assuming that the tool is correct?
What about dead code elimination that might be inhibited by adding an `assert` statement? Does that count as "generating new synthesizable hardware" ?

Reply:
Fixed